compose/foundation/foundation/src/desktopMain/kotlin/androidx/compose/foundation/ContextMenuProvider.desktop.kt

```diff
@@ -105,22 +106,26 @@ private val LocalContextMenuData = staticCompositionLocalOf<ContextMenuData?> {
  * @param key The pointer input handling coroutine will be cancelled and **re-started** when
  * [contextMenuOpenDetector] is recomposed with a different [key].
  * @param enabled Whether to enable the detection.
+ * @param pass The [PointerEventPass] to react to.
  * @param onOpen Invoked when a context menu opening event is detected, with the local offset it
- * should be opened at.
+ * should be opened at. Returns whether the event should be consumed.
  */
 @ExperimentalFoundationApi
 fun Modifier.contextMenuOpenDetector(
     key: Any? = Unit,
     enabled: Boolean = true,
-    onOpen: (Offset) -> Unit
+    pass: PointerEventPass = PointerEventPass.Main,
```

Comment:
I don't see a good reason for that. `Main` is the typical/default pass, and my use of `pass = Initial` in this PR is a special case.

Also, changing to a default value of `Initial` would break source-backwards-compatibility. Someone using `contextMenuOpenDetector { ... }` was getting his lambda called in the main pass, and after recompiling would get the call in the initial pass.

Reply:
Then the question can we avoid using `Initial` in our calling site? If it is only needed for us, and there is no use case for users, then extending opinionated gesture detector maybe not good.

I looked and the code more, it is used as a detector to select a word. This isn't fully correct, as `LocalTextContextMenu` can use own detector by design (override it to middle-click), and there will be strange behaviour - on right we select, on middle - we open the context menu.

To be correct, we need to use the old behaviour - reacting on `ContextMenuState.Status.Open`. I see that is changed because it is moved and we don't have access to the state. Let's discuss in a week, we need to resolve this somehow.

> changing to a default value of Initial would break source-backwards-compatibility

true, but it is still experimental